tests/integration/test_workflow_db_mgr.py

```diff
@@ -178,3 +179,39 @@ async def test_record_only_non_clock_triggers(
     assert db_select(schd, False, 'xtriggers', 'signature') == [
         ('xrandom(100)',),
         ('xrandom(100, _=Not a real wall clock trigger)',)]
+
+
+async def test_time_zone_writing(
+    one_conf,
+    flow,
+    scheduler,
+    start,
+    db_select,
+    set_timezone
+):
+    """Don't store scheduler startup timezone forever.
+
+    https://github.com/cylc/cylc-flow/issues/6701
+    """
+    set_timezone('XXX')
+    wid = flow(one_conf)
+    schd = scheduler(wid, paused_start=False, run_mode='live')
+    async with start(schd):
+        itask = schd.pool.get_tasks()[0]
+        schd.submit_task_jobs([itask])
+        set_timezone()
+        schd.task_events_mgr.process_message(itask, 'INFO', 'submitted')
+
+    # Check the db time_submit (defective) against time_submit_exit
+    # which was ok:
+    (time_submit, time_submit_exit), = db_select(
+        schd, False, 'task_jobs', 'time_submit', 'time_submit_exit'
+    )
+    time_submit = datetime.strptime(time_submit, '%Y-%m-%dT%H:%M:%S%z')
+    time_submit_exit = datetime.strptime(
+        time_submit_exit, '%Y-%m-%dT%H:%M:%S%z'
+    )
+
+    assert time_submit_exit >= time_submit
+    # The two times should be approx the same:
+    assert time_submit_exit < time_submit + timedelta(seconds=10)
```

Comment:
The use of `schd.task_events_mgr.process_message(itask, 'INFO', 'submitted')` is a bit artificial as it is just getting the current time, whereas normally this is called with a timestamp that comes from the jobs-submit context.

However, come to think of it, the `time_submit_exit` is not important, the fact of the matter is that the `time_submit` was bugged and that's all we need to test. So I would boil this test down to:

```suggestion
    set_timezone('XXX-19:00')
    schd = scheduler(flow(one_conf), paused_start=False, run_mode='live')
    async with start(schd):
        itask = schd.pool.get_tasks()[0]
        now = datetime.now().astimezone()
        set_timezone('XXX-19:17')
        schd.submit_task_jobs([itask])

    # Check the db time_submit:
    (time_submit,) = db_select(schd, False, 'task_jobs', 'time_submit')[0]
    time_submit = datetime.strptime(time_submit, '%Y-%m-%dT%H:%M:%S%z')
    # The submit time should be approx correct:
    assert (
        abs(time_submit - now) < timedelta(seconds=10)
    ), f"{time_submit} ~= {now}"
```